### PR TITLE
EDU: Basic attributes (migration, admin form)

### PR DESCRIPTION
**Pull request by eronisko**
image

@igor-kamil ~These JSON arrays work reasonably well... Do you think we really need the whole spatie package for keywords? I think we could do without it just fine for the keywords, at least for now.~ I've jumped the gun and implemented keywords without a library. Let me know what you think :)

### CODE REVIEW

File: app/Http/Controllers/ArticleController.php [hunk 1]
```diff
@@ -182,4 +177,30 @@ private function uploadMainImage($article)
...
+            'eduKeywordsOptions' => Article::all()
+                ->pluck('edu_keywords')
+                ->filter()
+                ->flatten()
+                ->unique()
+                ->sort()
+                ->reduce(function ($options, $value) {
+                    $options[$value] = $value;
+                    return $options;
+                 }) ?? collect(),
```
Comment (igor-kamil) [inline app/Http/Controllers/ArticleController.php:203]: 🥇 collection-fu

File: webpack.mix.js [hunk 2]
```diff
@@ -44,6 +44,7 @@ mix
     })
 
     // CSS
+    .less('resources/less/admin.less', 'public/css')
```
Comment (igor-kamil) [inline webpack.mix.js:47]: 👌

**Review by igor-kamil — APPROVED** [review]
it works nicely. there are some cosmetic comments - not blocking it from merging.

File: app/Article.php [hunk 3]
```diff
@@ -26,30 +27,50 @@ function getArtworksDirAttribute()
...
-    // public function items()
- //    {
- //        return $this->belongsToMany('Item', 'collection_item', 'collection_id', 'item_id');
- //    }
+    ];
+
+    protected $casts = [
+        'edu_media_types' => 'array',
+        'edu_target_age_groups' => 'array',
+        'edu_keywords' => 'array',
```
[the remark this answers is not in the document]
Comment (eronisko) [reply app/Article.php:43]: To be honest, I haven't found a case for the library so far... if we were to type/translate the tags then yes, but right now this just seems more straightforward to me.

File: app/Http/Controllers/ArticleController.php [hunk 4]
```diff
@@ -3,12 +3,10 @@
 namespace App\Http\Controllers;
 
 use App\Article;
-use Illuminate\Support\Facades\Validator;
+use Illuminate\Http\Request as HttpRequest;
```
[the remark this answers is not in the document]
Comment (eronisko) [reply app/Http/Controllers/ArticleController.php:6]: Good point, I'll replace it

File: database/migrations/2021_05_13_140453_add_edu_attributes_to_articles.php [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+     *
+     * @return void
+     */
+    public function up()
+    {
+        Schema::table('articles', function (Blueprint $table) {
+            $table->after('publish', function ($table) {
+                $table->json('edu_media_types')->nullable();
+                $table->json('edu_target_age_groups')->nullable();
+                $table->boolean('edu_suitable_for_home');
```
[the remark this answers is not in the document]
Comment (eronisko) [reply database/migrations/2021_05_13_140453_add_edu_attributes_to_articles.php:20]: >  what about to name it in the form of `is_` or `can_`

That would normally be my go-to preference, but this is more in-line with the other boolean attributes in the table (e.g. `promote`, `publish`).

> and last ->default(0) value?

0 is the default if the field is non-nullable, but I'll add `->default(0)` to be more explicit.

**Pull request merged**